flag_set_test.go

```diff
@@ -83,6 +83,7 @@ func TestFlagSet_Bool(t *testing.T) {
 		{args: []string{"--help"}, wantX: false, wantY: true, wantErr: ff.ErrHelp},
 		{args: []string{"--xflag", "-h"}, wantX: true, wantY: true, wantErr: ff.ErrHelp},
 		{args: []string{"-y", "--help"}, wantX: false, wantY: false, wantErr: ff.ErrHelp},
+		{args: []string{"--xflag=", "--help"}, wantX: true, wantY: false, wantErr: ff.ErrHelp},
```

Comment:
Shouldn't this be

```suggestion
		{args: []string{"--xflag=", "--help"}, wantX: true, wantY: true, wantErr: ff.ErrHelp},
```

?

Reply:
@peterbourgon I see what you mean: `yflag := fs.BoolDefault('y', "yflag", true, "another boolean flag")`. I didn't notice that `yflag` defaulted to true. Sorry.

That said, here's something weird: the tests pass without error in all the cases below.

+ `{args: []string{"--xflag=", "--help"}, wantX: true, wantY: true, wantErr: ff.ErrHelp}`
+ `{args: []string{"--xflag=", "--help"}, wantX: true, wantY: false, wantErr: ff.ErrHelp}`
+ `{args: []string{"--xflag=", "--help"}, wantX: true, wantErr: ff.ErrHelp}`

They all pass. I'm very confused. (I updated to `wantY: true`, but for some reason it doesn't seem to matter.)

Reply:
If wantErr is non-nil then the wantX and wantY fields aren't evaluated. My original comment was a bit pedantic, in that sense.

Reply:
Ah, okay. Either way, I clearly need to understand the test structure better.